eval.py

```diff
@@ -0,0 +1,91 @@
+
+import os
+import time
+import pickle
+import sys
+import argparse
+import json
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim
+from torch.utils.tensorboard import SummaryWriter
+from tqdm import tqdm
+from skimage import img_as_ubyte
+import imageio
+
+from models import Unet2D_simple as Unet2D
+from utils.data import sk_loader
+from utils.loss import WeightedFocalLoss, DiceLoss
+import utils.misc as workspace
+from torchvision.ops import sigmoid_focal_loss
+
+
+def eval(data_loader, model, exp_dir):
+    model.eval()
+    test_image_savepath = os.path.join(workspace.get_dir(exp_dir, workspace.evaluation_subdir), "images")
+    if not os.path.isdir(test_image_savepath):
+        os.makedirs(test_image_savepath)
+    pbar = tqdm(data_loader)
+    for _, data in enumerate(pbar):
+        img = data['image'].cuda()
+        # lab = data['mask'].cuda()
+        name = data['name']
+        outimg_f = os.path.join(test_image_savepath, f"testout_{name}.png")
+        with torch.no_grad():
+            inp_logits = model(img)
+            output = torch.sigmoid(inp_logits).data.cpu().numpy().squeeze()
+        imageio.imsave(outimg_f, img_as_ubyte(output))
+
+
+def main(args):
+    np.random.seed(2020)
+    torch.manual_seed(2020)
+    torch.cuda.manual_seed(2020)
+
+    if not os.path.isdir(args.exp_dir):
+        print(f"[ERROR] Experiment dir {args.exp_dir} does not exist!")
+        sys.exit(0)
```

Comment:
I don't foresee the use of main()  since trainer.py and eval.py can be thought of main programs themselves calling using the different functions and classes. But I agree that return(0) is a graceful way to exit the main function.